File: HtProject-sp-callback/src/com/database/LightModel/tbl_phone_locateItem.java
package com.database.LightModel;

public class tbl_phone_locateItem extends com.database.Logical.LightDataModel {

	public final static String identifyField = "id";
	public final static String tableName = "tbl_phone_locate";

	/** 主键值 */
	private Integer _id;

	/** 电话号码前缀 */
	private String _phone;
	/** tbl_city主键ID */
	private Integer _city_id;
	/** 运营商 */
	private Integer _operator;

	@Override
	public String IdentifyField() {
		return identifyField;
	}

	@Override
	public String TableName() {
		return tableName;
	}

	@Override
	protected String[] GetNullableFields() {
		return new String[] { null, "operator" };
	}

	public Boolean Is_operatorNull() {
		return IsNull(Fields.operator);
	}

	public void Set_operatorNull() {
		SetNull(Fields.operator);
	}

	public Integer get_id() {
		return this._id;
	}

	public void set_id(Integer value) {
		this._id = value;
	}

	/** 电话号码前缀 */
	public String get_phone() {
		return this._phone;
	}

	/** 电话号码前缀 */
	@SuppressWarnings("unused")
	public void set_phone(String value) {
		if (false && false)
			RemoveNullFlag(Fields.phone);
		else if (!false) {
			if (value == null)
				SetNullFlag(Fields.phone);
			else
				RemoveNullFlag(Fields.phone);
		}

		SetFieldHasUpdate(Fields.phone, this._phone, value);
		this._phone = value;
	}

	/** tbl_city主键ID */
	public Integer get_city_id() {
		return this._city_id;
	}

	/** tbl_city主键ID */
	@SuppressWarnings("unused")
	public void set_city_id(Integer value) {
		if (false && false)
			RemoveNullFlag(Fields.city_id);
		else if (!false) {
			if (value == null)
				SetNullFlag(Fields.city_id);
			else
				RemoveNullFlag(Fields.city_id);
		}

		SetFieldHasUpdate(Fields.city_id, this._city_id, value);
		this._city_id = value;
	}

	/** 运营商 */
	public Integer get_operator() {
		return this._operator;
	}

	/** 运营商 */
	@SuppressWarnings("unused")
	public void set_operator(Integer value) {
		if (false && true)
			RemoveNullFlag(Fields.operator);
		else if (!false) {
			if (value == null)
				SetNullFlag(Fields.operator);
			else
				RemoveNullFlag(Fields.operator);
		}

		SetFieldHasUpdate(Fields.operator, this._operator, value);
		this._operator = value;
	}

	/** 数据表字段列表对像 */
	public final class Fields {
		private Fields() {
		}

		public final static String id = "id";
		/** 主键 */
		public final static String PrimaryKey = "id";

		/** 电话号码前缀 */
		public final static String phone = "phone";
		/** tbl_city主键ID */
		public final static String city_id = "city_id";
		/** 运营商 */
		public final static String operator = "operator";

	}

}

File: xtone-dotnet-interface/n8wan.public/LightDataModel/tbl_phone_locateItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace LightDataModel
{
    /// <summary>
    /// tbl_phone_locate数据模型
    /// </summary>
    public partial class tbl_phone_locateItem : Shotgun.Model.Logical.LightDataModel
    {
        /// <summary>
        /// 数据表字段列表对像
        /// </summary>
        public sealed class Fields
        {
            private Fields() { }
            #region 表字段名称

            public const string id = "id";
            ///<summary>
            ///主键
            ///</summary>
            public const string PrimaryKey = "id";

            /// <summary>
            /// 电话号码前缀
            /// </summary>
            public const string phone = "phone";
            /// <summary>
            /// tbl_city主键ID
            /// </summary>
            public const string city_id = "city_id";
            /// <summary>
            /// 运营商
            /// </summary>
            public const string @operator = "operator";

            #endregion

        }
        #region 表字段变量定义
        /// <summary>
        /// 电话号码前缀
        /// </summary>
        private string _phone;
        /// <summary>
        /// tbl_city主键ID
        /// </summary>
        private int _city_id;
        /// <summary>
        /// 运营商
        /// </summary>
        private int _operator;

        #endregion

        public override string IdentifyField { get { return identifyField; } }
        public override string TableName { get { return tableName; } }

        public static readonly string identifyField = "id";
        public static readonly string tableName = "tbl_phone_locate";


        public int id
        {
            get;
            set;
        }

        #region 表字段值存取
        /// <summary>
        /// 电话号码前缀
        /// </summary>
        public string phone
        {
            get { return this._phone; }
            set
            {
#if false && false
				RemoveNullFlag(Fields.phone);
#elif !false
                if (value == null)
                    SetNullFlag(Fields.phone);
                else
                    RemoveNullFlag(Fields.phone);
#endif

                SetFieldHasUpdate(Fields.phone, this._phone, value);
                this._phone = value;
            }
        }
        /// <summary>
        /// tbl_city主键ID
        /// </summary>
        public int city_id
        {
            get { return this._city_id; }
            set
            {
#if true && false
				RemoveNullFlag(Fields.city_id);
#elif !true
			    if (value == null)
                    SetNullFlag(Fields.city_id);
                else
                    RemoveNullFlag(Fields.city_id);
#endif

                SetFieldHasUpdate(Fields.city_id, this._city_id, value);
                this._city_id = value;
            }
        }
        /// <summary>
        /// 运营商
        /// </summary>
        public int @operator
        {
            get { return this._operator; }
            set
            {
#if true && true
                RemoveNullFlag(Fields.@operator);
#elif !true
			    if (value == null)
                    SetNullFlag(Fields.@operator);
                else
                    RemoveNullFlag(Fields.@operator);
#endif

                SetFieldHasUpdate(Fields.@operator, this._operator, value);
                this._operator = value;
            }
        }

        #endregion
        #region 空值相关方法
        protected override string[] GetNullableFields()
        {
            return new string[]{null
			,"operator"
};
        }
        public bool IsoperatorNull() { return IsNull(Fields.@operator); }

        public void SetoperatorNull() { SetNull(Fields.@operator); }

        #endregion
        #region 静态方法
        /// <summary>
        /// 创建一个新的查询器
        /// </summary>
        public static Shotgun.Model.List.LightDataQueries<tbl_phone_locateItem> GetQueries(Shotgun.Database.IBaseDataClass2 dBase)
        {
            return new Shotgun.Model.List.LightDataQueries<tbl_phone_locateItem>(tableName, identifyField, dBase);
        }

        /// <summary>
        /// 创建一个新的查询器
        /// </summary>
        public static Shotgun.Model.List.LightDataQueries<tbl_phone_locateItem> GetQueries()
        {
            return new Shotgun.Model.List.LightDataQueries<tbl_phone_locateItem>(tableName, identifyField);
        }
        /// <summary>
        /// 根据主键查找指定的行,返回指定字段
        /// </summary>
        /// <param name="dBase"></param>
        /// <param name="id">主键值</param>
        /// <param name="fields">返回字段</param>
        /// <returns></returns>
        public static tbl_phone_locateItem GetRowById(Shotgun.Database.IBaseDataClass2 dBase, int id, string[] fields)
        {
            var q = GetQueries(dBase);
            q.Fields = fields;
            q.Filter.AndFilters.Add(identifyField, id);
            return q.GetRowByFilters();
        }

        /// <summary>
        /// 根据主键查找指定的行,返回所有字段
        /// </summary>
        /// <param name="dBase"></param>
        /// <param name="id">主键值</param>
        /// <returns></returns>
        public static tbl_phone_locateItem GetRowById(Shotgun.Database.IBaseDataClass2 dBase, int id)
        {
            return GetRowById(dBase, id, null);
        }

        #endregion
    }
}